Adds ysld unsupported module.

jdeolive opened the pull request:
YSLD is a Yaml based representation of YSLD with some syntactic nicities added to the mix. The module was developed by Boundless and is being contributed to the community. Thanks Boundless!

jodygarnett commented on modules/unsupported/ysld/pom.xml:
```diff
@@ -0,0 +1,100 @@
+  
+  <groupId>org.geotools</groupId>
+  <artifactId>gt-ysld</artifactId>
+  <name>Ysld Parser/Encoder</name>
```
Ylsd or YSLD :)

jodygarnett commented on modules/unsupported/ysld/pom.xml:
```diff
@@ -0,0 +1,100 @@
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <version>1.13</version>
```
Use of dependency management for version numbers, especially for things like hamcrest and findbugs that may conflict.

jodygarnett commented on modules/unsupported/ysld/src/main/java/org/geotools/ysld/Band.java:
```diff
@@ -0,0 +1,92 @@
+ *    http://geotools.org
+ *
+ *    (C) 2014-2016, Open Source Geospatial Foundation (OSGeo)
+ *    
```
This is one for @smithkm - should we have a (C) 2015-2016 Boundless to reflect this code has shipped in another product?

smithkm replied:
@jodygarnett I guess that comes down to that argument you had with Arne Kepp about the GWC headers.  I have no idea which is the correct way to do this to make the lawyers happy.

jdeolive commented:
Thanks @jodygarnett. Changed Ysld to YSLD (assuming that is the decided upon convention), added kevin and myself and module maintainers and moved dependency versions into dependency management section.

jodygarnett commented:
Added an updated readme, and changed the headers to show file history.
